test/models/user.rb

```diff
@@ -10,11 +10,17 @@ def initialize(client_ip:)
   end
 
   serialize :connection_details, Array
+  serialize :favourite_foods, Array
 
   attr_accessor :stored_transaction_changes
 
+  before_save :sort_favourite_foods
   after_commit :store_transaction_changes_for_tests
 
+  def sort_favourite_foods
+    self.favourite_foods.sort! if favourite_foods_changed?
+  end
```

Comment:
I think the test will be more readable if we do the changes explicitly in a transaction block, as you did for the first test you added, since it doesn't seem like it matters either way to the code under test.

Reply:
right, this is meant to test behaviour that may be unexpected to the developer (as what we're currently having with the introduction of the gem in BP). I'll add another test for the explicit one!